src/interfaces.ts

```diff
@@ -0,0 +1,22 @@
+export interface CartItem {
+  productId: string;
+  quantity: number;
+}
+
+export interface Cart {
+  email?: string;
+  items: CartItem[];
+};
+
+export type CartWorkflow = () => {
+  execute(): Promise<void>;
+  signals: {
+    addToCart: (item: CartItem) => void;
```

Comment:
You might like this syntax better:
```suggestion
    addToCart(item: CartItem): void;
```